src/aspire/source/image.py

```diff
@@ -802,6 +805,221 @@ def save_images(
+        # E[X**2] - E[X]**2
+        estimated_var = second_moment - first_moment**2
+        logger.debug(f"Source estimated signal var: {estimated_var}")
+
+        return estimated_var
+
+    def estimate_signal_power(
+        self,
+        sample_n=None,
+        support_radius=None,
+        batch_size=512,
+        signal_power_method="estimate_signal_mean_energy",
+    ):
+        """
+        Estimate the signal energy of `sample_n` projections using prescribed method.
+
+        :param sample_n: Number of images used for estimate.
+            Defaults to all images in source.
+        :param support_radius: Pixel radius used for masking signal support.
+            Default of None will compute inscribed circle, `self.L // 2`.
+        :param batch_size: Images per batch, defaults 512.
+        :param signal_power_method: Method used for computing signal energy.
+           Defaults to mean via `estimate_signal_mean_energy`.
+           Can use variance method via `estimate_signal_var`.
+
+        :returns: Estimated signal variance.
+        """
+
+        try:
+            signal_estimate_method = getattr(self, signal_power_method)
+        except AttributeError:
+            raise ValueError(
+                f"Cannot find signal_power_method={signal_power_method}."
+                "  Try the default 'estimate_signal_mean_energy' or 'estimate_signal_var'"
+            )
+
+        signal_power = signal_estimate_method(
+            sample_n=sample_n, support_radius=support_radius, batch_size=batch_size
+        )
+
+        return signal_power
+
+    def estimate_noise_power(
+        self,
+        sample_n=None,
+        support_radius=None,
+        batch_size=512,
+        noise_estimator=AnisotropicNoiseEstimator,
+    ):
+        """
+        Estimate the noise energy of `sample_n` images using prescribed estimator.
+
+        :param sample_n: Number of images used for estimate.
+            Defaults to all images in source.
+        :param support_radius: Pixel radius used for masking signal support.
+            Default of None will compute inscribed circle, `self.L // 2`.
+        :param batch_size: Images per batch, defaults 512.
+        :param noise_estimator: Method used for estimating noise.
+           Defaults to AnisotropicNoiseEstimator.
+
+        :returns: Estimated noise energy (variance).
+        """
+
+        if support_radius is None:
+            support_radius_proportion = 1
+        else:
+            # Note, noise_estimator expects radius as proportion.
+            support_radius_proportion = support_radius / (self.L // 2)
+
+        est = noise_estimator(
+            src=self, bgRadius=support_radius_proportion, batchSize=batch_size
+        )
+
+        return est.estimate()
+
+    def estimate_snr(
+        self,
+        sample_n=None,
+        support_radius=None,
+        batch_size=512,
+        noise_power=None,
+        signal_power_method="estimate_signal_mean_energy",
+        units=None,
```

Comment:
Is it common enough to denote SNR in cryo-EM using dB? I haven't seen it in any of the papers from the group, but maybe it occurs in others. All the same, it may make more sense to have the user do this separately (i.e., call `ratio_to_decibel`), if desired.

Reply:
Right, no, not in this group.  Image processing (including high noise arenas), I see it often, perhaps more common...

Makes sense to remove from the function (removed).  I'd like to leave the util method in there if that's alright.

Reply:
Looks good. Thanks.